desicochrane/basic-mailbox-api
# Add read and archive actions

**🤗commit-by-commit-friendly🤗**

# Why does this PR exist?

We need to be able to mark a message as read or archived via a http endpoint.

### How should this be manually tested?

1. `phpunit test`
1. make sure you have some messages loaded `php artisan mail-api:import <path to json>`
1. with curl/postman/your favorite tool mark a message as read/archived (remember to use basic auth with the admin token from your `.env` file)
    ```
    http://localhost/api/messages/1/read
    http://localhost/api/messages/1/archive
    ```
1. you can now either check in db or hit the api endpoint to confirm the message has been marked as read/archived

### Discussion

This is missing the "un-archive" and "un-read" actions
